boxoffice/views/admin_discount.py

```diff
@@ -0,0 +1,223 @@
+from ..models import db
+from boxoffice.models import Organization, DiscountPolicy, DiscountCoupon, Price, CURRENCY_SYMBOL, CURRENCY
+from ..forms import DiscountPolicyForm, DiscountCouponForm, DiscountPriceForm, CouponBasedDiscountPolicyForm, AutomaticDiscountPolicyForm, PriceBasedDiscountPolicyForm
+from utils import xhr_only, json_date_format, api_error, api_success
+
+
+def jsonify_price(price):
+    if price:
+        return {
+            'amount': price.amount,
+            'start_at': json_date_format(price.start_at),
+            'end_at': json_date_format(price.end_at)
+        }
+    else:
+        return None
+
+
+def jsonify_discount_policy(policy):
+    return {
+        'id': policy.id,
+        'name': policy.name,
+        'title': policy.title,
+        'discount_type': "Automatic" if policy.is_automatic else "Coupon based",
+        'item_quantity_min': policy.item_quantity_min,
+        'percentage': policy.percentage,
+        'is_price_based': policy.is_price_based,
+        'discount_code_base': policy.discount_code_base,
+        'bulk_coupon_usage_limit': policy.bulk_coupon_usage_limit,
+        'price_details': jsonify_price(Price.query.filter(Price.discount_policy == policy).first()) if policy.is_price_based else '',
+        'currency_symbol': CURRENCY_SYMBOL['INR'],
+        'dp_items': [{'id': str(item.id), 'title': "{ic_title}: {title}".format(ic_title=item.item_collection.title, title=item.title)} for item in policy.items]
+    }
+
+
+def jsonify_discount_policies(data_dict):
+    discount_policies_list = []
+    for policy in data_dict['discount_policies']:
+        discount_policies_list.append(jsonify_discount_policy(policy))
+    return jsonify(
+        org_name=data_dict['org'].name, title=data_dict['org'].title,
+        discount_policies=discount_policies_list,
+        currency=[currency for currency, label in CURRENCY.items()],
+        total_pages=data_dict['total_pages'],
+        paginated=data_dict['total_pages'] > 1,
+        current_page=data_dict['current_page']
+    )
+
+
+@app.route('/admin/o/<org>/discount_policy')
+@lastuser.requires_login
+@render_with({'text/html': 'index.html', 'application/json': jsonify_discount_policies})
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    permission='org_admin'
+    )
+@requestargs('search', ('page', int), ('size', int))
+def admin_discount_policies(organization, search=None, page=1, size=None):
+    results_per_page = size or 20
+
+    discount_policies = organization.discount_policies
+
+    if search:
+        discount_policies = discount_policies.filter(
+            DiscountPolicy.title.ilike('%{query}%'.format(query=search)))
+
+    total_policies = discount_policies.count()
+    total_pages = int(math.ceil(total_policies / results_per_page))
+    offset = (page - 1) * results_per_page
+
+    discount_policies = discount_policies.limit(results_per_page).offset(offset).all()
```

Comment:
Use `query.paginate`. [It handles the details for you](http://flask-sqlalchemy.pocoo.org/2.1/api/#flask.ext.sqlalchemy.BaseQuery.paginate).